scripts/rust-libc-config.yml

```diff
@@ -0,0 +1,2028 @@
+  - "cfsetspeed"
+  # unix/linux_like/mod.rs
+  - "sem_destroy"
+  - "sem_init"
+  - "fdatasync"
+  - "mincore"
+  - "clock_getres"
+  - "clock_gettime"
+  - "clock_settime"
+  - "clock_getcpuclockid"
+  - "dirfd"
+  - "pthread_getattr_np"
+  - "pthread_attr_getstack"
+  - "memalign"
+  - "setgroups"
+  - "pipe2"
+  - "statfs"
+  - "fstatfs"
+  - "memrchr"
+  - "posix_fadvise"
+  - "futimens"
+  - "utimensat"
+  - "duplocale"
+  - "freelocale"
+  - "newlocale"
+  - "uselocale"
+  - "mknodat"
+  - "pthread_condattr_getclock"
+  - "pthread_condattr_setclock"
+  - "pthread_condattr_setpshared"
+  - "pthread_mutexattr_setpshared"
+  - "pthread_rwlockattr_getpshared"
+  - "pthread_rwlockattr_setpshared"
+  - "ptsname_r"
+  - "clearenv"
+  - "waitid"
+  - "getresuid"
+  - "getresgid"
+  - "acct"
+  - "brk"
+  - "sbrk"
+  - "vfork"
+  - "setresgid"
+  - "setresuid"
+  - "wait4"
+  - "login_tty"
+  - "execvpe"
+  - "fexecve"
+  - "getifaddrs"
+  - "freeifaddrs"
+  - "bind"
+  - "writev"
+  - "readv"
+  - "sendmsg"
+  - "recvmsg"
+  - "uname"
+  - "strchrnul"
+
+forced_functions:
+  - "__errno_location"
+
+force_raw_function_pointer:
+  - "pthread_create"
+
+force_local_type:
+  - "option"
+
+force_struct_member_type:
+  "sigaction":
+    - name: "__sa_handler"
+      rename-to: "sa_sigaction"
+      type: "sighandler_t"
+  "siginfo_t":
+    - name: "__si_fields"
+      type: "[::c_char; 124]"
+  "stat":
+    - name: "st_atim"
+      replace:
+        - name: "st_atime"
+          type: "::time_t"
+        - name: "st_atime_nsec"
+          type: "i64"
+    - name: "st_mtim"
+      replace:
+        - name: "st_mtime"
+          type: "::time_t"
+        - name: "st_mtime_nsec"
+          type: "i64"
+    - name: "st_ctim"
+      replace:
+        - name: "st_ctime"
+          type: "::time_t"
+        - name: "st_ctime_nsec"
+          type: "i64"
+
+force_struct_zero_fill:
+  - "__mlibc_mutex"
+  - "__mlibc_cond"
+  - "__mlibc_fair_rwlock"
+
+force_macro_type:
+  "::c_ulong":
+    - "FIOCLEX"
+    - "FIONBIO"
+    - "FIONCLEX"
+    - "FIONREAD"
+    - "SIOCDEVPRIVATE"
+    - "SIOCGIFMTU"
+    - "SIOCSIFMTU"
+    - "TIOCGWINSZ"
+    - "TIOCSCTTY"
+    - "TIOCSWINSZ"
+  "::c_uint":
+    - "IGNBRK"
+    - "BRKINT"
+    - "IGNPAR"
+    - "PARMRK"
+    - "INPCK"
+    - "ISTRIP"
+    - "INLCR"
+    - "IGNCR"
+    - "ICRNL"
+    - "IUCLC"
+    - "IXON"
+    - "IXANY"
+    - "IXOFF"
+  "usize":
+    - "NCCS"
+    - "VEOF"
+    - "PTHREAD_STACK_MIN"
+  "::speed_t":
+    - "B0"
+    - "B50"
+    - "B75"
+    - "B110"
+    - "B134"
+    - "B150"
+    - "B200"
+    - "B300"
+    - "B600"
+    - "B1200"
+    - "B1800"
+    - "B2400"
+    - "B4800"
+    - "B9600"
+    - "B19200"
+    - "B38400"
+    - "B57600"
+    - "B115200"
+    - "B230400"
+    - "B460800"
+    - "B500000"
+    - "B576000"
+    - "B921600"
+    - "B1000000"
+    - "B1152000"
+    - "B1500000"
+    - "B2000000"
+    - "B2500000"
+    - "B3000000"
+    - "B3500000"
+    - "B4000000"
```

Comment:
Missing a newline at the end of file. We should fix that at the next opportunity but it's not a blocker for me.